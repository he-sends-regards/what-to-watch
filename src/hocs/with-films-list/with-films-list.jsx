import React from "react";

const withFilmsList = (Component) => (props) => {
  const [activeFilmId, setActiveFilmId] = React.useState(-1);

  const onMouseEnter = (evt) => {
    evt.preventDefault();
    setTimeout(setActiveFilmId(Number(evt.currentTarget.dataset.id), 1000));
  };

  const onMouseLeave = (evt) => {
    evt.preventDefault();
    setActiveFilmId(-1);
  };

  return (
    <Component
      {...props}
      activeFilmId={activeFilmId}
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
    >
    </Component>
  );
};

export {withFilmsList};
