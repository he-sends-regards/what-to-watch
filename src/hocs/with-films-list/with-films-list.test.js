import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import {withFilmsList} from "./with-films-list";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <React.Fragment>
      {children}
    </React.Fragment>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withFilmsList(MockComponent);

it(`WithFilmsList is rendered correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped>
      <React.Fragment/>
    </MockComponentWrapped>
  ), {
    createNodeMock() {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
