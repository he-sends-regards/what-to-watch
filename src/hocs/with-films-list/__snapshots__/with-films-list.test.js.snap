// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithFilmsList is rendered correctly 1`] = `null`;
